### ISSUE TITLE
find command adds colors to result also in scripts

### ISSUE THREAD

**Issue by sam701** [nushell/nushell#11899]
### Describe the bug

The `find` command adds coloring to the matched lines also in scripts. This feature breaks scripts.

### How to reproduce

```sh
> let a = ('https://www.nushell.sh' | find shell)
> http get $a
Error: nu::shell::unsupported_input

  × Unsupported input
   ╭─[entry #19:1:1]
 1 │ http get $a
   · ────┬─── ─┬
   ·     │     ╰── value: '"\u{1b}[37mhttps://www.nu\u{1b}[0m\u{1b}[41;37mshell\u{1b}[0m\u{1b}[37m.sh\u{1b}[0m"'
   ·     ╰── Incomplete or incorrect URL. Expected a full URL, e.g., https://www.example.com
```

### Expected behavior

I expected nu to return the page content.

Using `ansi strip` is a helpful workaround but I still think the commands when being invoked in scripts should not add coloring.

### Configuration

| key                | value                                                |
| ------------------ | ---------------------------------------------------- |
| version            | 0.90.1                                               |
| branch             |                                                      |
| commit_hash        |                                                      |
| build_os           | macos-aarch64                                        |
| build_target       | aarch64-apple-darwin                                 |
| rust_version       | rustc 1.75.0 (82e1608df 2023-12-21) (Homebrew)       |
| cargo_version      | cargo 1.75.0                                         |
| build_time         | 2024-02-06 22:28:49 +00:00                           |
| build_rust_channel | release                                              |
| allocator          | mimalloc                                             |
| features           | dataframe, default, extra, sqlite, trash, which, zip |
| installed_plugins  |                                                      |

**Comment by sholderbach** [comment]
I agree that the current behavior of `find` to insert ANSI styling is highly problematic if you want to use it in conjunction with any programmatic tools.

For example our file system operations all contain a bit of extra code to strip ANSI escape codes. While this makes them usable with sane path names and `find` this is not really sound behavior as path names are allowed to contain more legal bytes depending on the OS/FS (our ANSI stripping logic also strips e.g. `\t`)

A possible way forward would be to split the behavior of find into two parts. 

`find` itself performs the filtering and sends the used filter into the `PipelineMetadata` so that the `table` command which is implicitly added at the end of a pipeline when displaying, can use that to perform the highlighting of matches.

**Comment by NotTheDr01ds** [comment]
Pulling @sholderbach's comment from the doc repo directly in here so that it doesn't get lost.

When this is fixed, can someone please open an issue or PR to update [this example in the doc](https://github.com/nushell/nushell.github.io/pull/1694#discussion_r1894950720)? Thanks in advance ;-)
